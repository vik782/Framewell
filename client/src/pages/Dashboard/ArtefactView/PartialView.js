/**
 * @fileoverview The Partial View Component in the Dashboard. Shows
 *               slightly-more information of an artefact that was just
 *               clicked/tapped
 * Uses:
 * - React for rendering HTML
 * - Iconify for adding icons
 * - React Router for handling client-side routes
 */

// Imports of packages
import React, { useState } from "react";
import { Icon } from "@iconify/react";
import { Link } from "react-router-dom";

// Imports of local components
import Skewer from "../../../components/Skewer";

// Style-based imports
import "./PartialView.scss";

/**
 * The partial view takes in all the required information as props from its
 * parent component and displays them in a figure HTML tag
 * @return {React.Component}
 */
function PartialView({ title, image, desc, date, _id, openFunction, setUserData, currPageNum, setCurrPageNum, setNumPages, mode }) {

  return (
    <figure className="partial-view">
      <div className="exit-partial">
        <Icon icon="bi:x-lg" color="#f8f8f8" className="exit-button" onClick={()=>openFunction(_id)}/>
      </div>
      <div className="image-side">
        <img
          className="image-partial-view"
          src={image.imgURL}
          alt="the partial view of the component"
        />
      </div>
      <div className="info-side">
        <div className="info-values">
          <h2 className="info-title">
            {title}
          </h2>
          <div className="info-desc">
            <p>
              <span id="p-desc">Description:</span>
              <br/>
              <span className={'grey smaller'}>{(desc) ? desc : "No description given"}</span>
              
            </p>
          </div>
        </div>
        <div className="info-more">
          <Link to={`/full-view/${_id}`} className="link-line">
            <p>
              Click to Display Full View
              <Icon
                className="redirect-icon"
                icon="bi:box-arrow-in-down-right"
                flip="vertical"
              />
            </p>
          </Link>
        </div>
      </div>
      <div className="skewer-partial">
        <Skewer
          _id={_id}
          className="skewer-menu-partial"
          setUserData={setUserData}
          currPageNum={currPageNum}
          setCurrPageNum={setCurrPageNum}
          setNumPages={setNumPages}
          mode={mode}
        />
      </div>
    </figure>
  );
}

export default PartialView;
