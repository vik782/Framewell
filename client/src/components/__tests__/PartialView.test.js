import React from "react";
import PartialView from "../src/pages/Dashboard/PartialView";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

// Snapshot test to account for changes to the UI, whether expected or unexpected
test("Partial View renders correctly", () => {
    const args = { 
        title: "StoryBook Test",
        image: "https://www.ecosia.org/images?q=drop%20down%20like%20google%20images%20html%20css#id=8313FFDF3E9CDBC97F9A27492E8C441675DF2EF5",
        desc: "testing if storybook is working and if partial view is working",
        date: "12/02/02",
        _id: "/6325c0d2c579241489ad0bd0"
    };
    const partialView = renderer.create(<MemoryRouter><PartialView {...args}/></MemoryRouter>).toJSON();
    expect(partialView).toMatchSnapshot();
});