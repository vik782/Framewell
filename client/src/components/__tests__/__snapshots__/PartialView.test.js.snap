// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partial View renders correctly 1`] = `
<figure
  className="partial-view"
>
  <div
    className="image-side"
  >
    <img
      className="image-partial-view"
    />
  </div>
  <div
    className="info-side"
  >
    <div
      className="info-title"
    >
      <p>
        StoryBook Test
      </p>
    </div>
    <div
      className="info-dec"
    >
      <p>
        Description:
      </p>
      <br />
      <p>
        testing if storybook is working and if partial view is working
      </p>
    </div>
    <div
      className="info-date"
    >
      <p>
        12/02/02
      </p>
    </div>
    <div
      className="info-more"
    >
      <a
        className="link-line"
        href="//6325c0d2c579241489ad0bd0"
        onClick={[Function]}
      >
        <p>
          Click to Display Full View
          <span />
        </p>
      </a>
    </div>
  </div>
</figure>
`;
